src/main/java/io/sealights/plugins/sealightsjenkins/buildsteps/cli/CLIHandler.java

```diff
@@ -112,7 +112,8 @@ public void setCommandArgument(AbstractCommandArgument commandArgument) {
     private AbstractUpgradeManager getRelevantUpgradeManager(
             UpgradeProxy upgradeProxy, UpgradeConfiguration upgradeConfiguration) {
 
-        boolean useScannerUpgradeManager = CommandModes.Config.equals(commandArgument.getMode());
+        boolean useScannerUpgradeManager =
```

Comment:
Not sure that I get it. Mind elaborating?

Reply:
This is a very ugly code which coupled the upgrade logic and relevant jar, for config and prCoonfig modes we need the build scanner, all other uses test listener.